frontend/src/components/taskSelection/index.js

```diff
@@ -1,133 +1,114 @@
-import { fetchLocalJSONAPI } from '../../network/genericJSONRequest';
-
-
-class TaskSelection extends React.Component {
-  getProjectTasksPromise;
-  state = {
-    activeSection: 'tasks',
-    tasks: {},
-  };
+import { TaskList } from './taskList';
 
-  componentDidMount() {
-    if (this.props.project.projectId) {
-      this.getProjectTasks();
-    }
-  }
-  componentDidUpdate(prevProps) {
-    if (this.props.project.projectId && this.props.project.projectId !== prevProps.project.projectId) {
-      this.getProjectTasks();
-    }
-  }
-  getProjectTasks = event => {
-    this.getProjectTasksPromise = cancelablePromise(
-      fetchLocalJSONAPI(`projects/${this.props.project.projectId}/tasks/`)
-    );
-    this.getProjectTasksPromise.promise
-      .then(data => this.setState({tasks: data}))
-      .catch(e => console.log(e));
-  }
-  renderHeaderLine() {
+function HeaderLine({author, projectId, priority}: Object) {
     const userLink = (
-      <Link to={`/users/${this.props.project.author}`} className="link blue-dark underline">
-        {this.props.project.author}
+      <Link to={`/users/${author}`} className="link blue-dark underline">
+        {author}
       </Link>
     );
     return (
       <div className="cf">
         <div className="w-70 dib fl">
-          <span className="blue-light">#{this.props.project.projectId}</span>
+          <span className="blue-light">#{projectId}</span>
           <span className="blue-dark">
             {' '}
             | <FormattedMessage {...messages.createBy} values={{ user: userLink }} />
           </span>
         </div>
         <div className="mw4 dib fr">
-          <PriorityBox priority={this.props.project.projectPriority} extraClasses={'pv2 ph3'} />
+          <PriorityBox priority={priority} extraClasses={'pv2 ph3'} />
         </div>
       </div>
     );
   }
 
-  render() {
-    return (
-      <div>
-        <div className="cf pv3">
-          <div className="w-100 w-60-ns fl">
-            <div className="ph4">
-              <ReactPlaceholder
-                showLoadingAnimation={true}
-                rows={3}
-                delay={500}
-                ready={typeof(this.props.project.projectId) === 'number' && this.props.project.projectId > 0}
-              >
-                {this.renderHeaderLine()}
-                <div className="cf pb3">
-                  <h3 className="f2 fw6 mt2 mb3 ttu barlow-condensed blue-dark">
-                    {this.props.project.projectInfo && this.props.project.projectInfo.name}
-                  </h3>
-                  <span className="blue-light">{this.props.project.campaignTag} &#183; Brazil</span>
+export function TaskSelection({project, type, loading}: Object) {
+  const [activeSection, setActiveSection] = useState('tasks');
+  const defaultEditor = useSelector(state => state.preferences.default_editor);
+  const [error, tasksLoading, tasks] = useFetch(`projects/${project.projectId}/tasks/`);
```

Comment:
${project.projectId} can become `undefined` like with:
`http://localhost:3000/projects/1999/map`
https://tasks-stage.hotosm.org/projects/1999/map

There must be some other related issue too. You could check on this fetch so it cannot crash in other situations. Here is the localhost log (the order is not expected):
```
app           | 172.25.0.1 - - [04/Oct/2019 04:28:17] "OPTIONS /api/v2/projects/1999/queries/summary/ HTTP/1.1" 200 -
app           | 172.25.0.1 - - [04/Oct/2019 04:28:17] "OPTIONS /api/v2/projects/undefined/tasks/ HTTP/1.1" 200 -
app           | 172.25.0.1 - - [04/Oct/2019 04:28:18] "GET /api/v2/projects/undefined/tasks/ HTTP/1.1" 200 -
app           | [2019-10-04 04:28:18,134] CRITICAL in resources: Project Summary GET - unhandled error: 'NoneType' object is not iterable
app           | 172.25.0.1 - - [04/Oct/2019 04:28:18] "GET /api/v2/projects/1999/queries/summary/ HTTP/1.1" 500 -
```

Reply:
I added the not found content when the projectId doesn't exist. Thanks for noticing it!